gtsfm/averaging/rotation/cycle_consistency.py

```diff
@@ -1,343 +0,0 @@
-        v_corr_idxs_dict: dictionary, with key as image pair (i1,i2) and value as matching keypoint indices.
-        two_view_reports_dict: mapping from image pair indices (i1,i2) to a report containing information
-            about the verifier's output (and optionally measurement error w.r.t GT). Note: i1 < i2 always.
-        visualize: boolean indicating whether to plot cycle error vs. pose error w.r.t. GT
-
-    Returns:
-        i2Ri1_dict_consistent: subset of i2Ri1_dict, i.e. only including edges that belonged to some triplet
-            and had cycle error below the predefined threshold.
-        i2Ui1_dict_consistent: subset of i2Ui1_dict, as above.
-        v_corr_idxs_dict_consistent: subset of v_corr_idxs_dict above.
-        metrics_group: Rotation cycle consistency metrics as a metrics group.
-    """
-    cycle_errors = []
-    max_rot_errors = []
-
-    # (i1,i2) pairs
-    cycle_consistent_keys = set()
-
-    per_edge_errors = defaultdict(list)
-
-    # Extract edges that are ordered correctly and have a valid rotation.
-    valid_edges = []
-    for (i1, i2), i2Ri1 in i2Ri1_dict.items():
-        if i2Ri1 is None or i1 >= i2:
-            logger.error("Incorrectly ordered edge indices found in cycle consistency for ({i1}, {i2})")
-            continue
-        else:
-            valid_edges.append((i1, i2))
-
-    triplets = graph_utils.extract_cyclic_triplets_from_edges(valid_edges)
-
-    for (i0, i1, i2) in triplets:
-        cycle_error, max_rot_error, _ = compute_cycle_error(i2Ri1_dict, (i0, i1, i2), two_view_reports_dict)
-        # since i0 < i1 < i2 by construction, we preserve the property `a < b` for each edge (a,b)
-        per_edge_errors[(i0, i1)].append(cycle_error)
-        per_edge_errors[(i1, i2)].append(cycle_error)
-        per_edge_errors[(i0, i2)].append(cycle_error)
-
-        cycle_errors.append(cycle_error)
-        max_rot_errors.append(max_rot_error)
-
-    inlier_errors_aggregate = []
-    inlier_errors_wrt_gt = []
-
-    outlier_errors_aggregate = []
-    outlier_errors_wrt_gt = []
-
-    plt.close("all")
-    # aggregate info over per edge_errors
-    for (i1, i2), edge_cycle_errors in per_edge_errors.items():
-        if edge_acceptance_criterion == EdgeErrorAggregationCriterion.MIN_EDGE_ERROR:
-            error_aggregate = np.amin(edge_cycle_errors)
-
-        elif edge_acceptance_criterion == EdgeErrorAggregationCriterion.MEDIAN_EDGE_ERROR:
-            error_aggregate = np.median(edge_cycle_errors)
-
-        if error_aggregate < CYCLE_ERROR_THRESHOLD:
-            cycle_consistent_keys.add((i1, i2))
-            inlier_errors_aggregate.append(error_aggregate)
-            inlier_errors_wrt_gt.append(two_view_reports_dict[(i1, i2)].R_error_deg)
-        else:
-            outlier_errors_aggregate.append(error_aggregate)
-            outlier_errors_wrt_gt.append(two_view_reports_dict[(i1, i2)].R_error_deg)
-
-    if visualize:
-        plt.scatter(
```

Comment:
@akshay-krishnan I've found these scatter plots to be very useful, so I'd like to keep them

Reply:
@johnwlambert Do we want something out of this data in the report too? Things like precision and recall?

Reply:
added the plots and the precision recall metrics.